### DISCUSSION TITLE
Error codes are not exported anywhere

### DISCUSSION THREAD

**Discussion by mkermani144** [libp2p/js-libp2p#2669] [Q&A] [answered]
There are numerous error codes available in `libp2p` package, in `packages/libp2p/src/errors.ts`. Is there a reason these error code are not exported neither in the `interface` package, nor in the `libp2p` package itself?  
Currently there is no way to check caught error type against these codes other than duplicating the needed ones inside codebases using libp2p.  
I also know that these codes are [going to be refactored to error names](https://github.com/libp2p/js-libp2p/issues/2634) in the future, but it doesn't make any difference. When the refactor occurs, the names should be exported the same way.

**Comment by achingbrain** [comment, marked as answer]
`@libp2p/interface` exports `...Error` types reused by other libp2p modules and services.  These all have a static `.name` property that can be used to detect thrown errors.

Some modules define their own errors, for when they are too implementation-specific to include in `@libp2p/interface` but these should be exported by the relevant modules.  Sometimes things are missed though, so if they are not and you spot ones that are not exported, please open a PR to export them.

**Comment by mkermani144** [comment]
@achingbrain 
Yes, that seems to work. My main problem was about the `code` field in the previous version of `libp2p`, but having `name` property refactoring and considering the newly added error types, I think the issue is fixed now.  
By the way, I will open a PR if I find missing exports for the errors in any package. Thanks!
